airflow_dags/github-repos/import-data.pl

```diff
@@ -0,0 +1,204 @@
+    my $aHour = $1;
+    
+    return 0 unless ( $b =~ m/\d{4}-\d{2}-\d{2}-(\d+)/ );
+    my $bHour = $1;
+    
+    return $aHour <=> $bHour;
+}
+
+sub generate_inserts {
+    my $entries = shift;
+
+    open(SQL, '>', "updates.sql") or die $!;
+    
+    foreach my $entry ( @$entries ) {
+        my $base = $entry->{'payload'}{'pull_request'}{'base'}{'repo'};
+        unless ( $base ) {
+            warn "No base repo found, skipping";
+            next;
+        }
+        
+        die "no url name found for repo" unless $base->{'url'};
+
+        # Some fields changed their format name in 2015, so we look
+        # for multiple matches. The full_name property for repos isn't
+        # available until 2015 data, so we extract if from the URL in
+        # all cases.
+        my $name = quote_string(get_repo_name_from_url($base->{'url'}));
+        my $id = $base->{'id'};
+        my $description = quote_string($base->{'description'});
+        my $owner = quote_string($base->{'owner'}{'login'});
+        my $owner_type = quote_string($base->{'owner'}{'type'});
+        my $created_at = quote_string($base->{'created_at'});
+        my $updated_at = quote_string($base->{'updated_at'});
+        my $fork = $base->{'fork'} ? 'TRUE' : 'FALSE';
+        my $size = $base->{'size'} // 'NULL';
+        my $star_count = $base-> {'watchers'} // $base->{'stargazers_count'} // 'NULL';
+        my $fork_count = $base->{'forks'} // $base->{'forks_count'} // 'NULL';;
+        my $open_issues_count = $base->{'open_issues'} // $base->{'open_issues_count'} // 'NULL';
+        my $repo_language = quote_string($base->{'language'});
+        my $has_issues = $base->{'has_issues'} ? 'TRUE' : 'FALSE';
+        my $has_downloads = $base->{'has_downloads'} ? 'TRUE' : 'FALSE';
+        my $has_wiki = $base->{'has_wiki'} ? 'TRUE' : 'FALSE';
+        my $has_pages = $base->{'has_pages'} ? 'TRUE' : 'FALSE';
+        my $default_branch = quote_string($base->{'default_branch'});
+
+        print SQL <<"END";
+            REPLACE INTO repos 
+               (name, id, description, owner, owner_type, created_at, 
+                updated_at, fork,  size, star_count, fork_count, open_issues_count,
+                repo_language, has_issues, has_downloads, has_wiki, 
+                has_pages, default_branch)
+              VALUES ($name, $id, $description, $owner, $owner_type, $created_at, 
+                      $updated_at, $fork, $size, $star_count, $fork_count, $open_issues_count,
+                      $repo_language, $has_issues, $has_downloads, $has_wiki, 
+                      $has_pages, $default_branch);
+END
+    }
+
+    close(SQL)
+}
+
+sub get_repo_name_from_url {
+    my $url = shift;
+    if ( $url =~ m|/([^/]+)/([^/]+)$| ) {
+        return "$1/$2";
+    }
+    return undef
+}
+
+sub quote_string {
+    my $s = shift;
+    return 'NULL' unless defined $s;
+    $s =~ s/'/''/g;
+    $s =~ s/\\/\\\\/g;
+    return "'$s'";
+}
```

Comment:
Be warned. This approach will fuck you when the last character of a string is a '.

Reply:
Seems like it should work fine?